source/decisions/adrs/007-Integration-Approaches.html.md.erb

```diff
@@ -0,0 +1,61 @@
+---
+title: 007. Integration Approach
+weight: 2
+hide_in_navigation: true
+parent: /decisions
+last_reviewed_on: 2020-12-17
+review_in: 7 weeks
+---
+
+# 007. Integration Approach
+
+Date: 2020-12-17
+
+## Status
+
+For Review
+
+## Context
+
+The Children's Social Care Alpha Service is seeking to pull together key data assets
+such as Looked After Children and Foster Families. These will be sourced from Local
+Authority (LA) and Independent Foster Agency (IFA) Care Management Systems (CMS'),
+as well as central Departmental data collections such as SSDA903.
+
+The SSDA903 collection and the CMS at LAs highlight the significance of individual
+pathways through the Care System. Each individuals pathway can be regarded as a
+series of events and touchpoints that the individual has had with various other actors
+in the system, such as IFAs and Foster Families.
+
+The existing SSDA903 dataset is analogous to an event stream, with the Local
+Authorities (LAs) and Indedependent Fostering Agencies (IFAs) being event producers.
+
+## Assumptions
+
+* The Department's control over consumer environments, such as those run by LAs and IFAs,
+is very limited so it should go to a reasonable common denominator. The assumption here
+is that outbound access from their Social Care Systems to a REST API is possible in most situations.
+* A flat file interface will be offered in those cases where the above is not possible.
+* The [SSDA903 Technical Specification] can be used as a starting point for the events to collect.
+
+## Decision
```

Comment:
I think going the direction of Event Sourcing is definitely a good idea, we'll need to do some prototyping on it though. It's a great architectural setup, but not widely used I'd say (unfortunately).

I would definitely be slightly cautious about the ops setup in this approach. It is not trivial and we'll need some  DevOps help for sure.

Reply:
Totally agree about the caution. I am concerned about "premature optimisation" but I am struggling to dissuade myself from the view that this is worth attempting.